rbac/management/workspace/serializer.py

```diff
@@ -49,28 +48,32 @@ class Meta:
             "type",
         )
 
+    @property
+    def _service(self) -> WorkspaceService:
+        return self.context["view"]._service
+
     def create(self, validated_data):
         """Create the workspace object in the database."""
-        validated_data["tenant"] = self.context["request"].tenant
+        tenant = self.context["request"].tenant
+        return self._service.create(validated_data, tenant)
+
+    def validate(self, attrs):
+        from django.core.exceptions import ValidationError
+
+        """Validate on POST, PUT and PATCH."""
+        request = self.context.get("request")
+        parent_id = attrs.get("parent_id")
+        tenant = request.tenant
+
+        if parent_id and tenant:
```

Comment:
Should we leave this for now and add a ticket as @alechenninger mentioned to push this composite validation into the model?

Reply:
Per discussion, we can get rid of this in favor of 3 things (one will be TODO/new issue):

1. Meta-authz will check a user isn't trying to add a workspace to a parent they can't
2. The service should always pick the right tenant based on the parent workspace (using the user's org's default ws if no parent explicitly provided)
3. (TODO for later) Add a DB constraint that ensures a workspace's parent's tenant is always the same value (or null, to handle the root)